pkg/plugins/policies/core/xds/meshroute/clusters.go

```diff
@@ -105,7 +105,7 @@ func GenerateClusters(
 								// services that are synced can be accessed only with TLS through ZoneIngress
 								tlsReady = !ms.IsLocalMeshService() || ms.Status.TLS.Status == meshservice_api.TLSReady
 								if port, found := ms.FindPortByName(realResourceRef.Resource.SectionName); found {
-									protocol = port.AppProtocol
+									protocol = core_mesh.ParseProtocol(port.GetProtocol())
```

Comment:
why this here and not in the helpers?

Reply:
I couldn't return Protocol type from this method since I was getting circular dependencies errors on this. I can try maybe it could be entangled somehow

Reply:
I managed to untangle some of this mess. But there are still issues that we sometimes use plugin code or other stuff in core packages. I think targetRef is mostly to blame. Maybe this should be promoted to core now?

Reply:
> we sometimes use plugin code or other stuff in core packages

where in core packages we use plugin code?